oauth2/oauth2-resource-server/src/main/java/org/springframework/security/oauth2/server/resource/introspection/NimbusReactiveOpaqueTokenIntrospector.java

```diff
@@ -114,14 +118,35 @@ public NimbusReactiveOpaqueTokenIntrospector(String introspectionUri, WebClient
 	}
 
 	private Mono<HTTPResponse> adaptToNimbusResponse(ClientResponse responseEntity) {
+		MediaType contentType = responseEntity
+				.headers()
+				.contentType()
+				.orElseThrow(() -> {
+					this.logger.debug("Did not receive Content-Type from introspection endpoint in response");
+
+					throw new OAuth2IntrospectionException(
```

Comment:
```suggestion
					return new OAuth2IntrospectionException(
```
Since orElseThrow will throw the return value anyway